examples/ecdsa-psbt.rs

```diff
@@ -0,0 +1,629 @@
+        WatchOnly { account_0_xpub, input_xpub, master_fingerprint }
+    }
+
+    /// Creates the PSBT, in BIP174 parlance this is the 'Creater'.
+    fn create_psbt<C: Verification>(&self, secp: &Secp256k1<C>) -> Result<Psbt> {
+        let to_address = Address::from_str(RECEIVE_ADDRESS)?;
+        let to_amount = Amount::from_str(OUTPUT_AMOUNT_BTC)?;
+
+        let (_, change_address, _) = self.change_address(secp)?;
+        let change_amount = Amount::from_str(CHANGE_AMOUNT_BTC)?;
+
+        let tx = Transaction {
+            version: 2,
+            lock_time: 0,
+            input: vec![
+                TxIn {
+                    previous_output: OutPoint {
+                        txid: Txid::from_hex(INPUT_UTXO_TXID)?,
+                        vout: INPUT_UTXO_VOUT,
+                    },
+                    script_sig: Script::new(),
+                    sequence: Sequence::MAX, // Disable LockTime and RBF.
+                    witness: Witness::default(),
+                },
+            ],
+            output: vec![
+                TxOut {
+                    value: to_amount.to_sat(),
+                    script_pubkey: to_address.script_pubkey(),
+                },
+                TxOut {
+                    value: change_amount.to_sat(),
+                    script_pubkey: change_address.script_pubkey(),
+                }
+            ],
+        };
+
+        let psbt = Psbt::from_unsigned_tx(tx)?;
+
+        Ok(psbt)
+    }
+
+    /// Updates the PSBT, in BIP174 parlance this is the 'Updater'.
+    fn update_psbt(&self, mut psbt: Psbt) -> Result<Psbt> {
+        let mut input = Input { witness_utxo: Some(previous_output()), ..Default::default() };
+
+        let pk = self.input_xpub.to_pub();
+        let wpkh = pk.wpubkey_hash().expect("a compressed pubkey");
+
+        let redeem_script = Script::new_v0_p2wpkh(&wpkh);
+        input.redeem_script = Some(redeem_script);
+
+        let fingerprint = self.master_fingerprint;
+        let path = input_derivation_path()?;
+        let mut map = BTreeMap::new();
+        map.insert(pk.inner, (fingerprint, path));
+        input.bip32_derivation = map;
+
+        let ty = PsbtSighashType::from_str("SIGHASH_ALL").map_err(|_| Error::SighashTypeParse)?;
+        input.sighash_type = Some(ty);
+
+        psbt.inputs = vec![input];
+
+        Ok(psbt)
+    }
+
+    /// Finalizes the PSBT, in BIP174 parlance this is the 'Finalizer'.
+    fn finalize_psbt(&self, mut psbt: Psbt) -> Result<Psbt> {
+        use bitcoin::util::psbt::serialize::Serialize;
+
+        if psbt.inputs.is_empty() {
+            return Err(Error::InputsEmpty);
+        }
+
+        let sigs: Vec<_> = psbt.inputs[0].partial_sigs.values().collect();
+
+        let mut script_witness: Witness = Witness::new();
+        script_witness.push(&sigs[0].serialize());
+        script_witness.push(self.input_xpub.to_pub().serialize());
```

Comment:
nit: You can push pk from the `key` in the partial_keys map here instead. It seems more natural to use that instead of deriving it again.